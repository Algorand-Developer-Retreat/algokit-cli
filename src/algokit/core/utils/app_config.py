from pathlib import Path

import click

AppName = "algokit"


def get_config_dir():
    return Path(click.get_app_dir(AppName))


def read(file_name: str):
    return get_config_dir().joinpath(file_name).read_text("utf-8")


def write(file_name: str, content: str):
    get_config_dir().mkdir(parents=True, exist_ok=True)
    config_file = get_config_dir().joinpath(file_name)
    click.echo(click.style(f"Writing content to {config_file}", fg="blue"))
    config_file.write_text(content, "utf-8")
