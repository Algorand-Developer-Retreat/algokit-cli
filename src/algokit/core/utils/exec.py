import subprocess
from pathlib import Path

import click


def run(
    command: str,
    working_directory: Path | None = None,
    suppress_output: bool = False,
    throw_on_error: str | None = None,
):
    click.echo(click.style(f"Running '{command}' in '{working_directory or Path.cwd()}'", fg="blue"))
    result = subprocess.run(command.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd=working_directory)
    if not suppress_output and result.stdout.decode("utf-8").strip().__len__() > 0:
        click.echo(click.style(result.stdout.decode("utf-8"), fg="blue"))

    if throw_on_error is not None and result.returncode != 0:
        click.echo(click.style(result.stderr.decode("utf-8"), fg="red"))
        raise click.ClickException(throw_on_error)

    return result
