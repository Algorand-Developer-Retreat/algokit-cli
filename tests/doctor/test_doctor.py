from datetime import datetime, timezone

import pytest
from pytest_mock import MockerFixture
from utils.approvals import verify
from utils.click_invoker import invoke

FAKE_NOW = datetime(2020, 3, 11, 14, 0, 0)


@pytest.fixture(autouse=True)
def test_time_now(mocker: MockerFixture):
    mock_date = mocker.patch("algokit.core.doctor.datetime")
    fake_now = datetime(1990, 12, 31, 14, 0, 0, tzinfo=timezone.utc)
    mock_date.now.return_value = fake_now


def test_doctor_help(mocker: MockerFixture):
    result = invoke("doctor -h")

    assert result.exit_code == 0
    verify(result.output)


def test_doctor_default(mocker: MockerFixture):
    result = invoke("doctor")

    assert result.exit_code == 0
    verify(result.output)
