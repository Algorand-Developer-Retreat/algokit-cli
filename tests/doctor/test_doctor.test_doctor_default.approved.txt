DEBUG: Running 'pip show AlgoKit' in '{current_working_directory}'
DEBUG: pip: Name: algokit
DEBUG: pip: Version: 0.1.0
DEBUG: pip: Summary: Algorand development kit command line interface
DEBUG: pip: Home-page:
DEBUG: pip: Author: Algorand Foundation
DEBUG: pip: Author-email: contact@algorand.foundation
DEBUG: pip: License: MIT
DEBUG: pip: Location: {current_working_directory}/.venv/lib/python3.11/site-packages
DEBUG: pip: Requires: click, copier, httpx, questionary
DEBUG: pip: Required-by:
DEBUG: Running 'brew -v' in '{current_working_directory}'
DEBUG: brew: Homebrew 3.6.14
DEBUG: brew: Homebrew/homebrew-core (git revision d7bde1056b7; last commit 2022-12-07)
DEBUG: brew: Homebrew/homebrew-cask (git revision 6fb50d235a; last commit 2022-12-07)
DEBUG: Running 'docker -v' in '{current_working_directory}'
DEBUG: docker: Docker version 20.10.21, build baeda1f
DEBUG: Running 'docker-compose -v' in '{current_working_directory}'
DEBUG: docker-compose: Docker Compose version v2.12.2
DEBUG: Running 'git -v' in '{current_working_directory}'
DEBUG: git: git version 2.37.1 (Apple Git-137.1)
DEBUG: Running 'pipx --version' in '{current_working_directory}'
DEBUG: pipx: 1.1.0
DEBUG: Running 'poetry --version' in '{current_working_directory}'
DEBUG: poetry: Configuration file exists at /Users/mzaatar/Library/Application Support/pypoetry, reusing this directory.
DEBUG: poetry: 
DEBUG: poetry: Consider moving configuration to /Users/mzaatar/Library/Preferences/pypoetry, as support for the legacy directory will be removed in an upcoming release.
DEBUG: poetry: Poetry (version 1.2.2)
DEBUG: Running 'node -v' in '{current_working_directory}'
DEBUG: node: v18.12.1
DEBUG: Running 'npm -v' in '{current_working_directory}'
DEBUG: npm: 8.19.2
Time: 1990-12-31T22:00:00+08:00
AlgoKit: 0.1.0 {current_working_directory}/.venv/lib/python3.11/site-packages
Brew installed version: 3.6.14
OS: darwin 13.0.1
Docker: 20.10.21
Docker Compose: 2.12.2
Docker Compose 2.5 required to `run algokit sandbox command`; install via https://docs.docker.com/compose/install/Git: 2.37.1
[TODO] AlgoKit Python: <version of algokit version> <path> (base path: <base path>)
[TODO] Global Python: <version of first global path python3 version> <path> | None found
Pipx: 1.1.0
Poetry: 1.2.2
Node: 18.12.1
Npm: 8.19.2
If you are experiencing a problem with algokit, feel free to submit an issue via https://github.com/algorandfoundation/algokit-cli/issues/new; please include this output, if you want to populate this message in your clipboard, run `algokit doctor -c`
